### PR TITLE
Use proper C++ for ies classes

### PR DESCRIPTION
**Pull request by joakim-hove**
Use proper C++ for `ies::data::Data` and `ies::config::Config`

### CODE REVIEW

File: libres/lib/analysis/analysis_module.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ auto logger = ert::get_logger("analysis");
 
 struct analysis_module_struct {
     UTIL_TYPE_ID_DECLARATION;
-    ies::data::data_type *module_data;
+    std::unique_ptr<ies::data::Data> module_data;
```
Comment (pinkwah) [inline libres/lib/analysis/analysis_module.cpp:45]: Why `unique_ptr` and not just?
```suggestion
    ies::data::Data module_data;
```

Comment (joakim-hove) [reply libres/lib/analysis/analysis_module.cpp:45]: Good question - I'll update

Comment (pinkwah) [reply libres/lib/analysis/analysis_module.cpp:45]: No. `unique_ptr` is not a trivially-constructible object so it cannot be used in conjunction with `malloc` (undefined behaviour). Please change the alloc function to use `new`.

**Review by pinkwah — APPROVED** [review]
:tada: :rocket: :full_moon:

**Pull request merged**